cli/pa_analyze_command.go

```diff
@@ -413,6 +425,118 @@ func (cmd *PaAnalyzeCmd) checkLaggingStreamReplicas(r *archive.Reader) (checkSta
 	return Pass, nil
 }
 
+const cpuUsageThreshold = 0.9 // Warn if any server is using more than 90% of the available CPU
+// checkCpuUsage checks the CPU usage of all servers and alerts if any server is using more than 90% of the available CPU
+func (cmd *PaAnalyzeCmd) checkCpuUsage(r *archive.Reader) (checkStatus, error) {
+	serverTags := r.ListServerTags()
+	examples := newCollectionOfExamples(cmd.examplesLimit)
+
+	for _, serverTag := range serverTags {
+		serverName := serverTag.Value
+		var serverVarz server.Varz
+
+		if err := r.Load(&serverVarz, &serverTag, archive.TagServerVars()); errors.Is(err, archive.ErrNoMatches) {
+			cmd.logWarning("Artifact 'VARZ' is missing for server %s", serverName)
+			continue
+		} else if err != nil {
+			return Skipped, fmt.Errorf("failed to load VARZ for server %s: %w", serverName, err)
+		}
+
+		cpuUsageForASingleCore := serverVarz.CPU / float64(serverVarz.Cores)
```

Comment:
Kind of a bummer, what you calculate here is actually 'average utilization' among all cores.

Meaning if 1 core is pegged at 100% and the other 15 are idle, we will see <7% usage, and it's a false negative.

Not much to do about it i guess.

Reply:
Ya, unfortunately, we don't collect any other CPU information about the host except num of cores and total utilization.